app.js

```diff
@@ -358,4 +360,63 @@ const toggleNavBar = (route, data) => {
             hideAnimation();
         }
     });
-}
\ No newline at end of file
+}
+
+/**
+ * confirm the user's Firebase Auth email and phone data match the user's Firestore email and phone data.
+ * There's a 'gotcha' with magic links -the firebase auth profile is stripped of the phone number auth when a magic link is used for email login.
+ * The 'if' case below handles this. We check the firebase auth && firestore participant profiles for a phone match. If no match, and a phone number exists in the firestore participant profile, we write update the auth phone number to the firebase auth profile.
+ * The 'else if' case handles the following: We write firebase auth and firestore participant data separately, one after another. There's a chance the first write (Firebase Auth) succeeds and the second wite (Firestore) fails.
+ * This only costs an API call if the data is inconsistent since we hava access to both datapoints from app init. It otherwise only costs the time to run the check.
+ */
+const checkAuthDataConsistency = async (firebaseAuthEmail, firebaseAuthPhoneNumber, firestoreParticipantEmail, firestoreParticipantPhoneNumber) => {
```

Comment:
`await checkAuthDataConsistency(firebaseAuthUser.email ?? '', firebaseAuthUser.phoneNumber ?? '', data.data[conceptIdMap.firebaseAuthEmail] ?? '', data.data[conceptIdMap.firebaseAuthPhone] ?? '');`

I didn't touch on this from a prior review. For my curiosity, I noticed the `checkAuthDataConsistency` function includes return boolean values after the update functions. For example, `updateFirebaseAuthPhoneTrigger` and `updateFirebaseAuthPhoneTrigger` have return booleans. This is also the case in the catch block and after the `try... catch`. I assume adding the boolean values after the `return` keyword was for code readability.